project/frontend/src/pages/Header.js

```diff
@@ -1,24 +1,76 @@
 import './Header.css';
 
-import React from 'react';
+import React, { useState, useContext } from 'react';
 
-function Header(props) {
+import { AuthContext } from '../components/Authentication.js';
+import Modal from '../components/Modal.js';
+import Loading from '../components/Loading.js';
+
+function UserModal(props) {
   return (
-    <div id='header'>
-      {props.isUserLoggedIn ? (
-        <div>Pic</div>
-      ) : (
-        [
-          <button key='1' className='header-button'>
-            Create Account
-          </button>,
-          <button key='2' className='header-button'>
-            Log in
-          </button>,
-        ]
-      )}
-    </div>
+    <Modal
+      open={props.isModalOpen}
+      onDismiss={props.toggleModal}
+      top='64px'
+      right='16px'>
+      <img
+        className='profile-pic large'
+        src={props.user.getImageUrl()}
+        alt='Large profile.'
+      />
+      <h4 className='user-name'>{props.user.getName()}</h4>
+      <h5 className='user-email'>{props.user.getEmail()}</h5>
+      <button>Profile</button>
+      <br />
+      <button className='sign-out' onClick={props.signOut}>
+        Sign Out
+      </button>
+    </Modal>
   );
 }
 
+function Header(props) {
+  const context = useContext(AuthContext);
```

Comment:
Let's name this authContext to be clear (in case more contexts are used in the future).